### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

File: gap/ProsetDerivedMethods.gi [hunk 1]
```diff
@@ -4,6 +4,17 @@
 # Implementations
 #
 
+## FIXME: CategoryFilter := C -> HasIsThinCategory( C ) and IsThinCategory( C ) and CanCompute( C, "UniqueMorphism" )
```
Comment (zickgraf) [inline gap/ProsetDerivedMethods.gi:7]: What's the background of this `FIXME`, i.e. why would you like to check for ´UniqueMorphism`?

Comment (mohamed-barakat) [reply gap/ProsetDerivedMethods.gi:7]: For a first approximation see https://github.com/homalg-project/Locales/issues/71.

There are two sorts of thin categories:
* The morphism has void/empty/trivial MorphismDatum
* The morphism has a MorphismDatum which witnesses it being a morphism

In the first case, such thin categories should be able to define `UniqueMorphism` as the CAP operation mentioned in the above-mentioned issue.

Comment (zickgraf) [reply gap/ProsetDerivedMethods.gi:7]: I see. But why should `UniqueMorphism` not be installed in the second case (if there is an algorithm for computing such a witness from given source and range)?

Comment (mohamed-barakat) [reply gap/ProsetDerivedMethods.gi:7]: Yes, this `UniqueMorphism` should be called `UniqueMorphismWithoutGivenWitness` or the other one should be called `UniqueMorphismWithGivenWitness`. We are open for suggestions.

**Pull request merged**